x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -11,16 +11,29 @@ import (
 // TODO add unit tests for all the keepers
 
 // GetCurrentDataCommitment creates the latest data commitment at current height according to
-// the data commitment window specified
+// the data commitment window specified.
+// Assumes that this method is called when:
+//
+//	ctx.BlockHeight() != 0 && ctx.BlockHeight()%int64(k.GetDataCommitmentWindowParam(ctx)) == 0
 func (k Keeper) GetCurrentDataCommitment(ctx sdk.Context) (types.DataCommitment, error) {
-	beginBlock := uint64(ctx.BlockHeight()) - k.GetDataCommitmentWindowParam(ctx) + 1
-	// for a data commitment window of 400, the ranges will be: 1-400;401-800;801-1200
-	endBlock := uint64(ctx.BlockHeight())
-
 	if !k.CheckLatestAttestationNonce(ctx) {
 		return types.DataCommitment{}, types.ErrLatestAttestationNonceStillNotInitialized
 	}
 	nonce := k.GetLatestAttestationNonce(ctx) + 1
+	dcWindow := k.GetDataCommitmentWindowParam(ctx)
+	// for a data commitment window of 400, the ranges will be: 1-400;401-800;801-1200
+	var beginBlock, endBlock uint64
+	if ctx.BlockHeight() == int64(dcWindow) {
+		beginBlock = 1
+		endBlock = dcWindow
+	} else {
+		lastDCC, err := k.GetLastDataCommitment(ctx)
+		if err != nil {
+			return types.DataCommitment{}, err
+		}
+		beginBlock = lastDCC.EndBlock + 1
+		endBlock = beginBlock + dcWindow - 1
```

Comment:
will defer to author is they think this is simpler
```suggestion
		beginBlock = lastDCC.EndBlock + 1
		endBlock = lastDCC.EndBlock + dcWindow
```